### DISCUSSION TITLE
Appium flutter driver can interacts with Android OS elements

### DISCUSSION THREAD

**Discussion by subhanmansuri** [appium-userland/appium-flutter-driver#692] [General]
Hello, It is possible to do interaction with the Android OS Elements when we are using the Appium flutter driver can you plz help me out ?

**Comment by KazuCocoa** [comment]
Yes. Please check examples in https://github.com/appium/appium-flutter-driver?tab=readme-ov-file#context-management or test code

**Reply by subhanmansuri** [reply]
When we are changing the context to NATIVE_APP so will it work with the FlutterDriver ? or we have to initiate the AndroidDriver Also ??

caps.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android");
			caps.setCapability(MobileCapabilityType.PLATFORM_VERSION, "12.0");
//			caps.setCapability(MobileCapabilityType.DEVICE_NAME, "Galaxy Note10+");
			caps.setCapability(MobileCapabilityType.UDID, "ZD222H4Y2M");
			caps.setCapability(MobileCapabilityType.APP, "src/main/resources/app/app-debug.apk");

			// Additional settings
//			caps.setCapability(MobileCapabilityType.AUTOMATION_NAME, "UiAutomator2");
			caps.setCapability(MobileCapabilityType.AUTOMATION_NAME, "flutter");
			caps.setCapability(MobileCapabilityType.NEW_COMMAND_TIMEOUT, 20);
			caps.setCapability("unicodeKeyboard", true);
			caps.setCapability("resetKeyboard", true);
			caps.setCapability("javascriptEnabled", true);
			caps.setCapability("autoGrantPermissions", true);
			caps.setCapability("autoAcceptAlerts", true);

			// Do not reset app state
			caps.setCapability("noReset", true);
			caps.setCapability("fullReset", false);

			// Starts the appium server programmatically
			serviceAND = AppiumDriverLocalService.buildService(new AppiumServiceBuilder()
					.usingDriverExecutable(new File(APPIUM_NODE_PATH)).withAppiumJS(new File(APPIUM_JS_PATH))
					.withIPAddress(localhost).usingPort(portAndroid)
					// .usingAnyFreePort()
					.withArgument(GeneralServerFlag.BASEPATH, "/wd/hub")
					.withArgument(GeneralServerFlag.RELAXED_SECURITY).withArgument(GeneralServerFlag.SESSION_OVERRIDE)
					.withArgument(GeneralServerFlag.USE_DRIVERS, "flutter")
					.withArgument(GeneralServerFlag.LOG_LEVEL, "error")

			);

			serviceAND.start();
			Thread.sleep(2000);
			System.out.println("[ANDROID] Appium Server is running on port - " + portAndroid);
			driverAndroid = new AndroidDriver(serviceAND, caps);
			
			this is my setUp will it work can you plz check ? can you plz help me out ?

**Reply by KazuCocoa** [reply]
https://discuss.appium.io/ would help to ask general questions

**Comment by subhanmansuri** [comment]
When we are running the automation script the Login flow works fine but not interacting with the home page can you plz suggest some Solution  ? With FlutterDriver
